File: examples/Pascaltriangle/Pascaltriangle.rs


fn main()
{
	let n = 10;

	for k in 0..n 
	{
		for m in 0..k+1
		{
			print!("{}", pascaltriangle(k, m));
		}
		print!("\n");

	}
}

fn pascaltriangle(n:u32, i:u32) -> i64 {
	if n == i || i == 0 {
		return 1;
	}
	else{
		return pascaltriangle(n-1, i) + pascaltriangle(n-1, i-1);

	}
}

File: algo/Pascaltriangle/Pascaltriangle.c
#include <stdio.h>

long pascaltriangle(int, int);

int main()
{
 	int n = 10;

	for(int k = 0; n >= k; k++)
	{
		for(int s = 0; s < n-k; s++)
			printf(" ");
		for(int m = 0; k >= m; m++)
		{
			long f = pascaltriangle(k, m);
			printf("%ld ", f);
		}
		printf("\n");
	}
	return 0;
}

long pascaltriangle(int n, int i) 
{
	if(n == i || i == 0)
		return 1;
	else
		return pascaltriangle(n-1, i) + pascaltriangle(n-1, i-1);
}
